rest_framework/filters.py

```diff
@@ -85,6 +85,9 @@ def must_call_distinct(self, queryset, search_fields):
             opts = queryset.model._meta
             if search_field[0] in self.lookup_prefixes:
                 search_field = search_field[1:]
+            # Annotated fields do not need to be distinct
+            if isinstance(queryset, models.QuerySet) and search_field in queryset.query.annotations:
+                return False
```

Comment:
@andrzej3393 ,  what if we have several `search_fields`, and the first is an annotated field, and the others are m2m not annotated?

Reply:
Good catch. Opened #7146.